extensions/wikia/Recirculation/js/recirculation.js

```diff
@@ -3,183 +3,112 @@ require([
+		$mixedContentFooterContent.show();
+		require(['ext.wikia.recirculation.views.mixedFooter'], function (viewFactory) {
+			viewFactory().render({
+				nsItems: nsItems,
+				wikiItems: wikiItems,
+				discussions: discussions
 			});
-
-		// We need to keep track of the liftigniter helpers so we can setup tracking for them
-		liftigniterHelpers[recircModule.id] = configuredHelper;
-
-		if (recircModule.viewModule) {
-			views[recircModule.viewModule] = views[recircModule.viewModule] || [];
-			views[recircModule.viewModule].push(deferred);
-		}
+		});
 	});
 
-	$.each(views, function (viewModule, promises) {
-		var deferred = $.Deferred();
-
-		completed.push(deferred);
-
-		log('Initializing View: ' + viewModule, 'info', logGroup);
-		require([viewModule], function (viewFactory) {
-			$.when.apply($, promises)
-				.done(function () {
-					var view = viewFactory(),
-						args = Array.prototype.slice.call(arguments),
-						data = {
-							title: '',
-							items: []
-						};
-
-					log(args, 'info', logGroup);
+	// fetch data for all recirculation modules
+	liftigniter.fetch('ns');
 
-					args.forEach(function (result) {
-						if (!result) {
-							return;
-						}
+	var lazyLoadHandler = $.throttle(50, function () {
+			var mcfOffset = $mixedContentFooter.offset().top,
+				scrollPosition = $(window).scrollTop(),
+				windowInnerHeight = $(window).height(),
+				lazyLoadOffset = 500,
+				aproachingMCF = scrollPosition > mcfOffset - windowInnerHeight - lazyLoadOffset;
 
-						if (result.title && data.title.length === 0) {
-							data.title = result.title;
-						}
+			if (aproachingMCF) {
+				liftigniter.fetch('wiki');
+				discussions.fetch();
+				window.removeEventListener('scroll', lazyLoadHandler);
 
-						data.items = data.items.concat(result.items);
-					});
-
-					view.render(data)
-						.then(view.setupTracking())
-						.then(deferred.resolve);
-				})
-				.fail(handleError(viewModule), function () {
-					deferred.reject(viewModule);
-				});
-		});
+			}
 	});
 
-	$.when.apply($, completed)
-		.done(function () {
-			log('Finished rendering recirculation', 'info', logGroup);
+	window.addEventListener('scroll', lazyLoadHandler);
 
-			$.each(liftigniterHelpers, function (viewModule, helper) {
-				helper.setupTracking();
-			});
-		})
-		.fail(function (viewModule) {
-			log('Error running recirc at: ' + viewModule, 'info', logGroup);
-		});
-
-	function handleError(viewModule) {
-		return function (errorMessage) {
-			log(errorMessage, 'info', logGroup);
 
-			if (viewModule === 'ext.wikia.recirculation.views.premiumRail') {
-				require([
-					'ext.wikia.recirculation.helpers.fandom',
-					viewModule
-				], function (helper, view) {
-					helper({
-						type: 'community',
-						fill: true,
-						limit: 10
-					}).loadData()
-						.done(function (data) {
-							view().render(data);
-						});
-				});
-			}
-		};
-	}
+	// TODO handle errors
+	// TODO LI tracking
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

Reply:
what about this TODO?